libs/wingc/src/lsp/sync.rs

```diff
@@ -165,6 +178,14 @@ pub mod test_utils {
 			},
 		});
 
+		let diags = get_diagnostics();
+		// none of these should be panics
+		let panic_diags = diags
+			.iter()
+			.filter(|d| d.message.starts_with("Compiler bug"))
+			.collect::<Vec<_>>();
+		assert_eq!(panic_diags.len(), 0, "Compiler bug detected: {:#?}", panic_diags);
```

Comment:
I see this code reused in two places. One way to keep things DRY could be to put the logic in a trait I think (`impl Vec<Diagnostic> { ... }`)
```suggestion
		diags.assert_no_panics();
```

Reply:
ok I guess in Rust you can't just do `impl Vec<Diagnostic>` but you could do something like this:

```rust
pub trait DiagnosticVecExt {
	fn assert_no_bugs(&self);
}

impl DiagnosticVecExt for Vec<Diagnostic> {
	fn assert_no_panics(&self) {
		if !self.is_empty() {
			panic!("BUG: Diagnostics should be empty at this point");
		}
	}
}
```

naming the trait with "Ext" on the end is a Rust convention IIRC

Reply:
Deduping makes sense, but I just went the route of adding a fn to the diagnostics module. That way it was easy to throw `#[cfg(test)]` to reduce the binary size (this shouldn't be used outside of tests imo).